assayist/common/models/source.py

```diff
@@ -46,6 +46,10 @@ class SourceLocation(AssayistStructuredNode):
     component = RelationshipTo('Component', 'SOURCE_FOR', cardinality=ZeroOrOne)
     # Source locations that are embedded in this source location
     embedded_source_locations = RelationshipTo('SourceLocation', 'EMBEDS')
+    # The next version of this artifact
+    next_version = RelationshipFrom('SourceLocation', 'SUPERSEDES', cardinality=ZeroOrOne)
```

Comment:
I don't think the cardinality is right. 1.3 might precede both 1.4 and 2.0.
```
* 2.0
| * 1.4
|/
* 1.3
|
```

Reply:
@twaugh I don't think we should create edges to every previous version. We only need an edge between a version and the previous version. With that, we can traverse the graph to get older versions. For instance: `2.0` -> `1.4` -> `1.3` -> `1.2`.

It's a more complex query, but it saves us from creating millions of extra edges, and the maintenance is easier when a new version is added to the graph.

Reply:
I agree with keeping it a linked list, not a 1-to-many map. If we wanted the latest version link back to every previous version then we'd wind up with an exponential explosion of edges. Traversing the graph is what graph dbs are good at, let's just neo4j do its job.